internal/oci/runtime_vm.go

```diff
@@ -453,7 +462,8 @@ func (r *runtimeVM) execContainerCommon(ctx context.Context, c *Container, cmd [
 			return -1, killErr
 		}
 		<-execCh
-		return -1, errors.Errorf("ExecSyncContainer timeout (%v)", timeoutDuration)
+		// do not make an error for timeout: report it with a specific error code
+		return -2, nil
```

Comment:
Please, define an error instead of just using "-2",

Reply:
Done. I'm keeping it as a separate commit for now, just tell me if it's ok, and then I can squash if needed.

Reply:
I think I prefer this in two separate commits, thanks @littlejawa!